truth_set/python/structure/solution-approach/semantic-duplication/utils/dateUtils.py

```diff
@@ -0,0 +1,11 @@
+def format_date(date, format='YYYY-MM-DD'):
+    year = date.year
+    month = f"{date.month:02d}"
+    day = f"{date.day:02d}"
+    
+    if format == 'MM/DD/YYYY':
+        return f"{month}/{day}/{year}"
+    elif format == 'DD-MM-YYYY':
+        return f"{day}-{month}-{year}"
+    else:
+        return f"{year}-{month}-{day}"
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider using built-in date formatting</b></div>
<div id="fix">

Consider using Python's built-in `datetime.strftime()` method instead of implementing custom date formatting logic. The current implementation duplicates functionality already available in the standard library.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<blockquote>Check the AI-generated fix before applying</blockquote>
<div id="code">


```suggestion
def format_date(date, format='YYYY-MM-DD'):
    format_map = {
        'YYYY-MM-DD': '%Y-%m-%d',
        'MM/DD/YYYY': '%m/%d/%Y',
        'DD-MM-YYYY': '%d-%m-%Y'
    }
    strftime_format = format_map.get(format, '%Y-%m-%d')
    return date.strftime(strftime_format)
```

</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/rkj6124/bito-test/pull/1#issuecomment-2536353661>#c621b6</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged